File: basic_hash/basic.py
# Given a string, find the most commonly occurring letter.

# Input: “peter piper picked a peck of pickled peppers”
# Output: “p”


def basic(string):
    i = 0
    letters = {}
    mostCount = 0
    mostLetters = 0

    while i < len(string):
        if letters.get(string[i]):
            letters[string[i]] += 1
        else:
            letters[string[i]] = 1

        if letters[string[i]] > mostCount:
            mostCount = letters[string[i]]
            mostLetters = string[i]
        i += 1

    return mostLetters

print(basic("peter piper picked a peck of pickled peppers"))

File: basic_hash/basic.rb
# Given a string, find the most commonly occurring letter.

# Input: “peter piper picked a peck of pickled peppers”
# Output: “p”

def basic(string)

  letters = {}
  i = 0
  most_count = 0
  most_letters = ""

  while i < string.length
    if letters[string[i]]
      letters[string[i]] += 1
    else
      letters[string[i]] = 1
    end
    if letters[string[i]] > most_count
      most_count = letters[string[i]]
      most_letters = string[i]
    end
    i += 1  
  end

  return most_letters
end

p basic("peter piper picked a peck of pickled peppers")